tests/post_deploy_functional_readonly/main_test.go

```diff
@@ -31,7 +31,15 @@ func TestFunctionAppModule(t *testing.T) {
 		SetTestConfig(&testimpl.ThisTFModuleConfig{}).
 		SetTestConfigFolderName(testConfigsExamplesFolderDefault).
 		SetTestConfigFileName(infraTFVarFileNameDefault).
+		SetTestSpecificFlags(map[string]types.TestFlags{
+			"complete": {
+				"IS_TERRAFORM_IDEMPOTENT_APPLY": false,
+			},
+			"private_func_app": {
+				"IS_TERRAFORM_IDEMPOTENT_APPLY": false,
+			},
+		}).
 		Build()
 
-	lib.RunNonDestructiveTest(t, *ctx, testimpl.TestFunctionApp)
+	lib.RunNonDestructiveTest(t, *ctx, testimpl.TestFunctionApp, testimpl.TestPrivateFuncApp)
```

Comment:
in this read-only test, since you are testing only the private_fuction_app, I would suggest you remove testimpl.TestFunctionApp from this function args.